src/main/java/com/tozny/sdk/realm/methods/user_push/UserPushRequest.java

```diff
@@ -0,0 +1,54 @@
+package com.tozny.sdk.realm.methods.user_push;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.tozny.sdk.ToznyApiRequest;
+
+import javax.annotation.Nullable;
+
+/**
+ * Collects request parameters for invoking the "realm.user_push" method.
+ * Requires that either the `user_id` or `email` or `username` field be set.
+ */
+@JsonAutoDetect(getterVisibility=JsonAutoDetect.Visibility.NONE)
+public class UserPushRequest implements ToznyApiRequest {
+
+    @JsonProperty private String method = "realm.user_push";
+    @JsonProperty private String session_id;
+    @JsonProperty private String user_id;
+    @JsonProperty private String email;
+    @JsonProperty private String username;
+
+    public UserPushRequest(String session_id, @Nullable String user_id, @Nullable String email, @Nullable String username) {
```

Comment:
I think a Builder would be really useful for this class. Having four string arguments can be confusing and easy to screw up. Instead, make this constructor private and have a Builder that calls it. 

Also improves error handling and will make the overloads you define in a a later commit cleaner.